File: templates/problems/19_acm_io/06_graph_input/graph_input.go
package graph_input

import "bufio"

func solve(reader *bufio.Reader, writer *bufio.Writer) {
	// Read n nodes, m edges. Then m lines of "u v" edges.
	// Output the out-degree of each node (1 to n), one per line.
}

// ACM main() template:
// package main
//
// import (
// 	"bufio"
// 	"fmt"
// 	"os"
// )
//
// func solve(reader *bufio.Reader, writer *bufio.Writer) {
// 	// your solution here
// }
//
// func main() {
// 	reader := bufio.NewReader(os.Stdin)
// 	writer := bufio.NewWriter(os.Stdout)
// 	defer writer.Flush()
// 	solve(reader, writer)
// }

File: templates/problems/19_acm_io/06_graph_input/python/graph_input.py
import sys


def solve(input_data: str) -> str:
    """Read a directed graph with n nodes and m edges, output the out-degree
    of each node (1 to n), one per line.

    ACM-style: input format is:
        n m
        u1 v1
        u2 v2
        ...
        um vm

    Output is n lines: the out-degree of node i for i = 1..n.
    """
    # Parse n, m, then m edges. Count out-degree per node (1-indexed).
    #
    # 适用题型：图的度数/连通性、BFS/DFS遍历、拓扑排序、Dijkstra最短路
    # 典型输入：第一行 n m，接着 m 行每行 u v 表示一条边
    return ""


# For actual ACM usage:
# if __name__ == "__main__":
#     data = sys.stdin.read()
#     print(solve(data), end="")
